pkg/virt-handler/virtwrap/api/converter.go

```diff
@@ -48,6 +49,51 @@ func Convert_v1_Disk_To_api_Disk(diskDevice *v1.Disk, disk *Disk) error {
 	return nil
 }
 
+func diskBusFromMachine(bus, machine string) string {
+	if bus != "" {
+		// use what it was explicitely selected
+		return bus
+	}
+	// catches: "q35", "pc-q35-*"
+	// see /path/to/qemu-kvm -machine help
+	if strings.HasPrefix(machine, "pc-q35") || strings.HasPrefix(machine, "q35") {
+		return "sata"
+	}
+	// safe fallback for x86_64, but very slow
+	return "ide"
+}
+
+func diskDeviceFromBusAndIndex(deviceName, bus string, index int) string {
+	if deviceName != "" {
+		return deviceName
+	}
+	prefix := ""
+	switch bus {
+	case "virtio":
+		prefix = "vd"
+	case "sata", "scsi":
+		prefix = "sd"
+	case "ide":
+		prefix = "hd"
+	default:
+		// we should never get there
+		return ""
```

Comment:
I feel that this should be an error condition. If this code block is unexpected then failing silently is not the ideal choice--we'd end up with a disk named `a`... :)

If there's at least a reasonable default (for example "hd"?) this should still be logged somewhere.